msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/ADAP3DFeatureDetectionMethod.java

```diff
@@ -149,4 +188,22 @@
     return featureList;
 
   }
+
+  /** {@inheritDoc} */
+  @Override
+  public Float getFinishedPercentage() {
+    return objPeakDetection.getFinishedPercent();
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<Feature> getResult() {
+    return finalFeatureList;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void cancel() {
+    objPeakDetection.cancel();
```

Comment:
I don't get it.can you please explain it? My understanding is if the cancel is called execute method of peakdetection will return null and if it'll return null then execute method ADAP3DFeatureDetectionMethod will also return null. So I think code flow won't continue.